federation-2/supergraph/src/command/compose.rs

```diff
@@ -13,7 +15,7 @@ pub struct Compose {
     ///
     /// NOTE: Each subgraph entry MUST contain raw SDL
     /// as the schema source.
-    config_file: Utf8PathBuf,
+    config_file: Option<Utf8PathBuf>,
```

Comment:
we could add an `input_type: InputType` argument here that is

```rust
enum InputType {
  Yaml
  Json
}
```
and then have methods on it like

```rust
impl InputType {
  fn read_input_from_file(&self, config_input: &FileDescriptorType) -> Result<SupergraphConfig> {
    let contents = config_input.read_file_descriptor()?;
    match &self {
      Self::Yaml => SupergraphConfig::new_from_yaml_file(&contents),
      Self::Json => SupergraphConfig::new_from_json_file(&contents)
    }
  }
}
```

That `new_from_json_file` is new as well, but would look similar to the existing yaml one, it would just call `serde_json::from_str` instead of `serde_yaml::from_str`

make the default be yaml for compatibility w/old versions of rover

```
impl Default for InputType {
  fn default() -> Self {
    Self::Yaml
  }
}
```